Sources/curvelib/secp256k1/PublicKey.swift

```diff
@@ -1,109 +0,0 @@
-//
-//  PublicKey.swift
-//  
-//
-//  Created by CW Lee on 12/01/2024.
-//
-#if canImport(tkeylibcurve)
-    import tkeylibcurve
-#endif
-
-import Foundation
-
-public extension Secp256k1 {
```

Comment:
can we wrap all the api under class/ extesion of secp256k1  (which can access via `curvelib.secp256k1`)
so that we have room to add different lib or function under different class if we want in the future
even if we are not adding it is still fine as it declaring the publickey is from secp256k1 lib instead just curvelib

Reply:
AFAIK this is just to replace secp256k1 for swift due to the gigabitcoin and gigabitcoin mess. In the end this will be pretty limited in scope since most others are already included in Apples cryptokit by default, they just chose not to do secp256k1 specifically.

If we want to be able to add new libraries here, we should split secp256k1 off of this into it's own library, then import it as a dependency here and export multiple products from this library, this is the standard way of doing this.

I wouldn't use an extension though since these are base classes and not extensions of existing classes.

Reply:
https://docs.swift.org/swift-book/documentation/the-swift-programming-language/extensions/